### ISSUE TITLE
Rarity is causing BlizzMovePlugin_Rematch to not function (?)

### ISSUE THREAD

**Issue by brett1337** [WowRarity/Rarity#511]
See here for context: https://github.com/Numynum/BlizzMovePlugin_Rematch/issues/1#issuecomment-1320993588

Is it possible to prevent Rarity from stopping rematch from moving?

**Comment by Duckwhale** [comment]
Uhm, no idea why Rarity should affect those frames in any way. My first guess would be a bug in Blizzard's new UI (wouldn't be the first), possibly related to UI taint? I'll need a taint log to see if that could be a problem.

See https://github.com/WowRarity/Rarity/issues/499 for instructions on how to generate a taint log, with only the minimal number of addons enabled if you would :)

**Comment by brett1337** [comment]
[taint.log](https://github.com/WowRarity/Rarity/files/10050332/taint.log)

Here you go, hope this helps.

**Comment by Duckwhale** [comment]
I don't see any source of taint being caused by Rarity in that log file.

It's probably just another bug in Blizzard's new UI, like all the others that have recently been reported...

**Comment by Numynum** [comment]
Author of BlizzMove, and the BM-Rematch plugin here 🙂

TL;DR, I'd suggest closing the issue here, as it's not _really_ Rarity's fault :)

I decided to look into it a bit more; and it seems to be an interesting interaction of
1.  Rematch removing all hooks, and setting their handlers for MouseDown roughly when the loading screen ends (`SetScript` on `PLAYER_ENTERING_WORLD`)
2. BlizzMove hooking MouseDown when Blizzard_Collections addon loads (`HookScript`)

this usually goes fine, because the Blizzard_Collections addon doesn't load until you open the UI, so Rematch sets up their stuff, and then BlizzMove does their thing afterwards, all good 👌
but then, when Rarity is enabled
1. Rarity force loads the Blizzard_Collections addon before `PLAYER_ENTERING_WORLD`
2. BlizzMove hooks MouseDown (because the addon is loaded; before `PLAYER_ENTERING_WORLD`)
3. Rematch nukes all hooks, when it finally loads, after `PLAYER_ENTERING_WORLD`

which leaves it in an immovable state 🙁

to be honest, this is something that's mostly caused by Rematch loading their things at an odd timing, and then nuking all handlers, instead of using HookScript (which is usually fine, since it's their own frames; but it does break plugins like mine)

I'm thinking of creating a workaround to detect and restore BlizzMove's handlers whenever an addon tries to nuke them, to prevent this issue, and similar issues in the future (this is the second time something like this came up; though this time it's a little less straight-forwards 🙂)

**Comment by brett1337** [comment]
I would never dream of blaming rarity ;). Interesting interaction and I can confirm it is fixed. Thanks!
